pkgs/games/prismlauncher/default.nix

```diff
@@ -51,7 +55,10 @@ stdenv.mkDerivation rec {
     quazip
     ghc_filesystem
     tomlplusplus
-  ] ++ lib.optional (lib.versionAtLeast qtbase.version "6") qtwayland;
+  ]
+  ++ lib.optional (lib.versionAtLeast qtbase.version "6") qtwayland
+  ++ lib.optional withMangohud mangohud;
```

Comment:
MangoHud is only a runtime dependency.

6.x just looks for the vulkan layer configuration and uses the library path defined there.

Though I personally only tested this when using the mangohud module from home-manager.

Make sure that mangohud is added to `XDG_DATA_DIRS`.
See relevant code: https://github.com/PrismLauncher/PrismLauncher/blob/1f0fc61b53898f90936adcb45e5834fee55fd143/launcher/MangoHud.cpp#L52